**В университет на факультеты A и B поступило равное количество
студентов, а на факультет C студентов поступило столько же, сколько на A
и B вместе. Вероятность того, что студент факультета A сдаст первую
сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7,
а для студента факультета C - 0.9. Студент сдал первую сессию. Какова
вероятность, что он учится:**  

                a) на факультете A;  
                б) на факультете B;  
                в) на факультете C?

In [1]:
import numpy as np
from math import factorial

Пусть $n$ - количество студентов на факультете $A$, тогда на факультете
$B$ учится $n$ студентов, а на факультете $C$ $2n$ студентов.  
Введем события $S_A, S_B, S_C$, что сессию сдаёт студент факультетов
$A$, $B$, и $С$ соответственно, и событие $D$-сессия сдана.  
события $S_A$, $S_B$, и $S_С$ образуют полную группу событий,
$\Rightarrow P(S_A)+P(S_B)+P(S_C)=1$  
T.e.
$\frac{1}{n}+\frac{1}{n}+\frac{1}{2n}=1\\n=2.5\\\Rightarrow P(S_A)=\frac{1}{n}=0.25, P(S_B)=\frac{1}{n}=0.25, P(S_C)=\frac{1}{2n}=0.5$  
Условные вероятности Наступления события $D$, при условии что произошло
одно из событий $S_n$ даны в условии задачи.  
Полная вероятность наступления события $D$ исходя из формулы **полной**
вероятности:  
$P(D)=0.25\cdot 0.8+0.25\cdot 0.7+0.5\cdot 0.9$

In [2]:
PD=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события D, P(D) = {PD}.')

Полная вероятность наступления события D, P(D) = 0.825.


По формуле **Байеса** вычислим вероятность того, на каком факультете
учится студент, успешно сдавший сессию.  
$P_D(S_A)=\frac{P(S_A)\cdot P_{S_A}(D)}{P(D)}$ студент учится на
факультете А  
$P_D(S_B)=\frac{P(S_B)\cdot P_{S_B}(D)}{P(D)}$ студент учится на
факультете В  
$P_D(S_C)=\frac{P(S_C)\cdot P_{S_C}(D)}{P(D)}$ студент учится на
факультете С

In [ ]:
PDSA=0.25*0.8/PD
PDSB=0.25*0.7/PD
PDSC=0.5*0.9/PD
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


------------------------------------------------------------------------

#### Ответ:

Вероятность того, что студент учится на факультете А:
$P_D(S_A)=0.2424$ или $\approx24.24\%$;  
Вероятность того, что студент учится на факультете B:
$P_D(S_B)=0.2121$ или $\approx21.21\%$;  
Вероятность того, что студент учится на факультете C:
$P_D(S_C)=0.5455$ или $\approx54.55\%$.

------------------------------------------------------------------------